markdown/util.py

```diff
@@ -44,7 +44,269 @@
+            DeprecationWarning,
+        )
+        # In-place extension should update both list and set.
+        for element in elements:
+            if element not in self._list:
+                self._list.append(element)
+        self._set.update(elements)
+
+    def index(self, value, start: int = 0, stop: int = sys.maxsize, /):
+        warnings.warn(
+            "Using block level elements as a list is deprecated, use it as a set instead.",
+            DeprecationWarning,
+        )
+        return self._list.index(value, start, stop)
+
+    def insert(self, index: int, element: str, /) -> None:
+        warnings.warn(
+            "Using block level elements as a list is deprecated, use it as a set instead.",
+            DeprecationWarning,
+        )
+        # In-place insertion should update both list and set.
+        self._list.insert(index, element)
+        self._set.add(element)
+
+    def intersection(self, *others: set[str]) -> set[str]:
+        # User expects a set back.
+        return self._set.intersection(*others)
+
+    def intersection_update(self, *others: set[str]) -> None:
+        # In-place intersection should update both list and set.
+        self._set.intersection_update(*others)
+        self._list.clear()
+        self._list.extend(sorted(self._set))
+
+    def isdisjoint(self, other: set[str], /) -> bool:
+        return self._set.isdisjoint(other)
+
+    def issubset(self, other: set[str], /) -> bool:
+        return self._set.issubset(other)
+
+    def issuperset(self, other: set[str], /) -> bool:
+        return self._set.issuperset(other)
+
+    def pop(self, index: int = -1, /) -> str:
+        # In-place pop should update both list and set.
+        element = self._list.pop(index)
+        self._set.remove(element)
+        return element
+
+    def remove(self, element: str, /) -> None:
+        # In-place removal should update both list and set.
+        self._list.remove(element)
+        self._set.remove(element)
+
+    def reverse(self) -> None:
+        warnings.warn(
+            "Using block level elements as a list is deprecated, use it as a set instead.",
+            DeprecationWarning,
+        )
+        self._list.reverse()
+
+    def sort(self, /, *, key: Callable | None = None, reverse: bool = False) -> None:
+        warnings.warn(
+            "Using block level elements as a list is deprecated, use it as a set instead.",
+            DeprecationWarning,
+        )
+        self._list.sort(key=key, reverse=reverse)
+
+    def symmetric_difference(self, other: set[str], /) -> set[str]:
+        # User expects a set back.
+        return self._set.symmetric_difference(other)
+
+    def symmetric_difference_update(self, other: set[str], /) -> None:
+        # In-place symmetric difference should update both list and set.
+        self._set.symmetric_difference_update(other)
+        self._list.clear()
+        self._list.extend(sorted(self._set))
```

Comment:
Same shortcut, though we should maintain list order by first removing then adding elements, probably.